addons/website/static/src/js/tours/tour_utils.js

```diff
@@ -443,26 +443,39 @@ function selectElementInWeSelectWidget(
 /**
  * Switches to a different website by clicking on the website switcher.
  *
- * @param {number} websiteId - The ID of the website to switch to.
  * @param {string} websiteName - The name of the website to switch to.
  * @returns {Array} - The steps required to perform the website switch.
  */
-function switchWebsite(websiteId, websiteName) {
-    return [{
+function switchWebsite(websiteName) {
+    const steps = [{
         content: `Click on the website switch to switch to website '${websiteName}'`,
         trigger: '.o_website_switcher_container button',
+    }, {
+        content: "Fetch the website's id",
+        trigger: `.o_website_switcher_container .dropdown-item:contains("${websiteName}")`,
+        run: function () {
+            // Infer the website's id from the switch website menu
+            const websiteId = this.$anchor[0].dataset.websiteId
+            if (websiteId === undefined) {
+                console.error(`Could not find a website with name ${websiteName}`)
+            } else {
+                steps[2].extra_trigger = steps[2].extra_trigger.replace('{websiteId}', websiteId.toString())
+                steps[3].trigger = steps[3].trigger.replace('{websiteId}', websiteId.toString())
```

Comment:
I really would avoid to do that @Xavier-Do, there is no guarantee that `steps[2]` still refers to the right step. I could very well imagine the tour framework do something like this when registering a tour:

```javascript
steps = givenSteps.map(step => {
    return Object.assign({
        // default values here
    }, step);
});
```

I would not trust the given steps *descriptions* are the actual steps *data* that the system will use.

Can't we just keep the current system and maybe review things python-side? No way to force the id of the tested website maybe? Or transmit the new id to the tour itself?

Reply:
@qsm-odoo up

Reply:
We can't do any of this in stable anyway, we don't remove function parameters in 16.0, so you probably should create a new method to handle this if really needed. But given the discussion in the meeting today, do we still need this? Won't tours be started on non-demo data, so you can just create the website on the python side and rely on its hardcoded ID on the JS side? 

(Note: Side-effect of the cookie thing: this cannot be used anymore to create an user tour so this really should be a dedicated method (maybe an extension of switch theme or something :shrug: ).

Reply:
Sure about the stable thing, we probably won't be applying the no demo data change in stable regardless so this is more to start somewhere with this PR.

Hardcoded IDs don't allow the test to be re-ran (psql sequences are not rollbacked when the transaction is) which is also an issue when developing the tests.

I will make a separate function for test tours to not disturb the function's signature in stable but bear in mind that function can not be used in user tours anyways because you can not guarantee neither the name nor the id of the website (it is only used in both hardcoded id tours, at least in 16.0).

Reply:
> Hardcoded IDs don't allow the test to be re-ran (psql sequences are not rollbacked when the transaction is) which is also an issue when developing the tests.

Oh, did not know, good to know. In any case, I agree that hardcoding the ID js side is bad... but in stable I would be cautious about changing things like that.

> bear in mind that function can not be used in user tours anyways because you can not guarantee neither the name nor the id of the website

You can never guarantee anything in an user tour. Step 1 "drag and drop the snippet at the bottom of the page", the user drags it at the top... then the tour may be stuck forever.  Same in this hypothetical case: "Create website "test"" on a clean database, you can just have step 2 to rely on a "test" website with ID 2 to have been created but... maybe the user just called it "hello" indeed, then the tour is dead, too bad :shrug: Anyways, we'll never make such a tour to explain the users how to switch websites anyway, this was just about being cautious with changing any JS function signature in stable :+1: 

Thanks for checking :+1:

Reply:
Hi @qsm-odoo 

Could you approve the current version of the fix, so that we can move forward with this work?
Thanks